lib/apiservers/engine/backends/container.go

```diff
@@ -506,3 +663,185 @@ func (c *Container) imageExist(imageID string) (storeName string, err error) {
+
+		// Swagger wants an io.reader so give it the reader pipe.  Also, the swagger call
+		// to set the stdin is synchronous so we need to run in a goroutine
+		go func() {
+			setStdinParams := interaction.NewContainerSetStdinParamsWithContext(inContext).WithID(name)
+			setStdinParams = setStdinParams.WithRawStream(stdinReader)
+			plClient.Interaction.ContainerSetStdin(setStdinParams)
+		}()
+
+		// Copy the stdin from the CLI and write to a pipe.  We need to do this so we can
+		// watch the stdin stream for the detach keys.
+		go func() {
+			var err error
+			if tty {
+				_, err = attach.CopyEscapable(stdinWriter, clStdin, keys)
+			} else {
+				_, err = io.Copy(stdinWriter, clStdin)
+			}
+
+			if err != nil {
+				log.Errorf(err.Error())
+			}
+			errors <- err
+
+			if stdinOnce && !tty {
+				// Close the stdin connection.  Mimicing Docker's behavior.
+				// FIXME: If we close this stdin connection and request it again,
+				// does the port layer reconnects the stdin?  Is this diff from
+				// current Docker behavior?
+				stdinReader.Close()
+			} else {
+				// Shutdown the client's request for stdout, stderr
+				errCancel()
+				outCancel()
+			}
+			wg.Done()
+		}()
+	}
+
+	if clStdout != nil {
+		wg.Add(1)
+		transport := httptransport.New(PortLayerServer(), "/", []string{"http"})
+		plClient := client.New(transport, nil)
+		transport.Consumers["application/octet-stream"] = httpkit.ByteStreamConsumer()
+		transport.Producers["application/octet-stream"] = httpkit.ByteStreamProducer()
+
+		// Swagger -> pipewriter.  Synchronous, blocking call
+		go func() {
+			getStdoutParams := interaction.NewContainerGetStdoutParamsWithContext(outContext).WithID(name)
+			_, err := plClient.Interaction.ContainerGetStdout(getStdoutParams, clStdout)
+			if clStdin != nil {
+				// Close the client stdin connection (e.g. CLI)
+				clStdin.Close()
+				inCancel()
+			}
+			if err != nil {
+				if _, ok := err.(*interaction.ContainerGetStdoutNotFound); ok {
+					errors <- derr.NewRequestNotFoundError(fmt.Errorf("No such container: %s", name))
+					wg.Done()
+					return
+				}
+
+				// If we get here, most likely something went wrong with the port layer API server
+				errors <- derr.NewErrorWithStatusCode(fmt.Errorf("Unknown error from the interaction port layer: %s", err),
+					http.StatusInternalServerError)
+				wg.Done()
+				return
+			}
+
+			errors <- nil
+			wg.Done()
+		}()
+	}
+
+	if clStderr != nil {
+		wg.Add(1)
+		transport := httptransport.New(PortLayerServer(), "/", []string{"http"})
+		plClient := client.New(transport, nil)
+		transport.Consumers["application/octet-stream"] = httpkit.ByteStreamConsumer()
+		transport.Producers["application/octet-stream"] = httpkit.ByteStreamProducer()
+
+		// Swagger -> pipewriter.  Synchronous, blocking call
+		go func() {
+			getStderrParams := interaction.NewContainerGetStderrParamsWithContext(errContext).WithID(name)
```

Comment:
Same go routine comment.  Please take this out of the routine and do this in the parent function.